### DISCUSSION TITLE
markup: <img> tag align="left", align="right" not working since 09-10-24, was fine 09-09-24. align="center" is unaffected (it's ok).

### DISCUSSION THREAD

**Discussion by fxkrait** [community/community#138339] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

On any rendered markup file (`.md`, `.org`, etc), the `align="left"` and the `align="right"` on `<img>` tags are not working as of today (09-10-24). They were working fine yesterday (09-09-24), and obviously before then as well. `align="center"` is still working fine, as before (not affected).
- Issue: https://github.com/github/markup/issues/1877

### An example with a recent wayback-machine cached render (09-04-24):
- [README.md](https://github.com/doomemacs/doomemacs/blob/master/README.md) has a `align="right"` inside an `<img>` tag.
```
  <img src="https://raw.githubusercontent.com/doomemacs/doomemacs/screenshots/cacochan.png" align="right" />
```
From wayback-machine of [09-04-24](https://web.archive.org/web/20240904212016/https://github.com/doomemacs/doomemacs
), it looks OK:

![09-04-24-wayback-doom](https://github.com/user-attachments/assets/5a73a19f-f2d9-434b-a9d2-18913b8a4477)

Currently, of [09-10-24](https://github.com/doomemacs/doomemacs
), it's not rendering correctly:

![09-10-24-doom](https://github.com/user-attachments/assets/764eb759-a7cf-42b7-ab6b-22046154b0dc)

-------

### And here's an older wayback-machine, but more generic example (05-04-24):
If we compare a [gist](https://gist.github.com/DavidWells/7d2e0e1bc78f4ac59a123ddf8b74932d), that showcases how image allignment works, to it's wayback machine cached render from [05-04-24](https://web.archive.org/web/20240503121836/https://gist.github.com/DavidWells/7d2e0e1bc78f4ac59a123ddf8b74932d), we can see that `align="left"` and the `align="right"`  are currently not working:

-  `align="left"` is currently not rendering correctly:
![09-10-24-left](https://github.com/user-attachments/assets/e5040d0b-fc2d-4b2a-b651-cc3a9fedc7c0)

- It looks fine on wayback machine.
![05-03-24-wayback-left](https://github.com/user-attachments/assets/aef45afe-b511-4af6-8cb9-4717951ada87)

-  `align="right"` is currently not rendering correctly:
![09-10-24-right](https://github.com/user-attachments/assets/4d8d6ce6-5ecd-464c-ab36-351509a5860c)

- It looks fine on wayback machine.
![05-03-24-wayback-right](https://github.com/user-attachments/assets/ccf6baa7-221c-43b3-8100-87ab13a1e71a)
 
Issue: https://github.com/github/markup/issues/1877

**Comment by ClaraCrazy** [comment]
This isnt the only issue with images since today. See [my profile readme](https://github.com/ClaraCrazy#-github-stats) for example:
![image](https://github.com/user-attachments/assets/f7c2d24b-ffde-4c1e-86b6-df72818d8bdb)

How it is supposed to look like, screenshot from archive.org:
![image](https://github.com/user-attachments/assets/ae15cf07-e7c8-4abd-a27e-10dc5541c70a)

Lets hope they fix this quickly

**Comment by szapp** [comment]
I have [added](https://github.com/github/markup/issues/1877#issuecomment-2342787536) to the referenced issue that also SVG images are no longer rendered and the images span full width despite being scaled down.

This seems quite critical because it breaks the README of GitHub repositories that often act as documentation.

**Comment by MatinT-SA** [comment]
I've also faced the same isssue since this morning where I made use of width with percentage values for my images to be displayed on my GitHub README.md file and unfortunately I just realized it's a bug from GitHub team since I'm not the only person having this issue.

The interesting part is that the issue only persists with percentage values like `width="48%"` but if you use pixel values (numbers), it works fine. like `width="200"`

Also I figured out there are a couple of alternative ways for specifying the width of graphs and stats like giving `card_width` as a **query parameter** to our API URLs like this:

`https://github-readme-stats.vercel.app/api?username=MatinT-SA&show_icons=true&theme=radical&card_width=200`

So I'm gonna use one of these methods temporarily until they fix the issue.

**Comment by fxkrait** [comment, marked as answer]
Will close this, as the `align="left"` and `align="right"` has since been fixed for `<img>` tags.

**Reply by MatinT-SA** [reply]
> Will close this, as the `align="left"` and `align="right"` has since been fixed for `<img>` tags.

```
<div>
  <img align="right" src="https://github-readme-stats.vercel.app/api?username=MatinT-SA&layout=compact&show_icons=true&theme=react" alt="MatinT-SA github-readme-stats" />
  <img align="left" src="https://streak-stats.demolab.com?user=MatinT-SA&theme=react&layout=compact" alt="MatinT-SA github-streaks-readme-stats" />
</div>
```

This is the code that I have and here's the result I get

![github img issue](https://github.com/user-attachments/assets/dc086e67-629e-4580-9e99-a7f215565dc0)

It seems at least it hasn't solved the problem for me or probably getting it wrong somewhere.

**Reply by fxkrait** [reply]
@MatinT-SA Just copy what @ClaraCrazy did in her [readme](https://github.com/ClaraCrazy/ClaraCrazy/blob/slut/README.md). She's got a similar layout to what you're trying to do, and her horizontal layout has since been fixed to display properly, the same as it was before this issue ever occurred. 
- There's also a `width` element you can set in each `<img>`, but just look at that linked readme for real examples using that.
``` code
<div align="center">
  <img src="https://streak-stats.demolab.com?user=MatinT-SA&theme=react&layout=compact" alt="MatinT-SA github-streaks-readme-stats"/>
  <img src="https://github-readme-stats.vercel.app/api?username=MatinT-SA&layout=compact&show_icons=true&theme=react" alt="MatinT-SA github-readme-stats"/>
</div>
```
![image](https://github.com/user-attachments/assets/390b53e4-bd2b-48f5-a180-6cb01524bba3)

**Reply by MatinT-SA** [reply]
> @MatinT-SA Just copy what @ClaraCrazy did in her [readme](https://github.com/ClaraCrazy/ClaraCrazy/blob/slut/README.md). She's got a similar layout to what you're trying to do, and her horizontal layout has since been fixed to display properly, the same as it was before this issue ever occurred.
> 
> * There's also a `width` element you can set in each `<img>`, but just look at that linked readme for real examples using that.
> 
> ```
> <div align="center">
>   <img src="https://streak-stats.demolab.com?user=MatinT-SA&theme=react&layout=compact" alt="MatinT-SA github-streaks-readme-stats"/>
>   <img src="https://github-readme-stats.vercel.app/api?username=MatinT-SA&layout=compact&show_icons=true&theme=react" alt="MatinT-SA github-readme-stats"/>
> </div>
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/7407672/368344508-390b53e4-bd2b-48f5-a180-6cb01524bba3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2NDMyNzksIm5iZiI6MTcyNjY0Mjk3OSwicGF0aCI6Ii83NDA3NjcyLzM2ODM0NDUwOC0zOTBiNTNlNC1iZDJiLTQ4ZjUtYTE4MC02Y2IwMTUyNGJiYTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MThUMDcwMjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGYzZjMzN2YxNDFlNzlhMzdlMTA5ZjcwOTUxZDU1YTBiYjcxMWE3ZGM0MDdjMWRjODBiMjNlODA3YmZhNmJjNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Ly9ZYKt12Ono-BQS5Do5ycCVNhL-VZ6a_Gjy3O_rPrU)


Oh, I just checked it right now and realized the issue with using the width attribute is solved, and everything is back to normal. Anyway, thanks for your help.

```
<div>
  <img align="center" src="https://streak-stats.demolab.com?user=MatinT-SA&theme=react&layout=compact" alt="MatinT-SA github-streaks-readme-stats" width="51%" />
  <img align="center" src="https://github-readme-stats.vercel.app/api?username=MatinT-SA&layout=compact&show_icons=true&theme=react" alt="MatinT-SA github-readme-stats" width="48%" />
</div>
```
